editor/Storyboarding/EditorGeneratorContext.cs

```diff
@@ -51,10 +51,10 @@ public EditorGeneratorContext(Effect effect, string projectPath, string mapsetPa
             this.watcher = watcher;
         }
 
-        public override StoryboardLayer GetLayer(string identifier)
+        public override StoryboardLayer GetLayer(string identifier, OsbLayer osbLayer = OsbLayer.Background)
         {
             var layer = EditorLayers.Find(l => l.Identifier == identifier);
-            if (layer == null) EditorLayers.Add(layer = new EditorStoryboardLayer(identifier, effect));
+            if (layer == null) EditorLayers.Add(layer = new EditorStoryboardLayer(identifier, effect, osbLayer));
```

Comment:
For what it's worth, it *does* mirror the behavior default values for configurable variables in an effect. That is, for an existing effect, changing the default value thankfully does not overwrite an effect's configurable setting.

I do think there's a bit of ambiguity and a lack of giving full trust to the method. Me passing an argument specifying the specific layer something should go on may not actually happen, without me realizing that. At least layers are boxed to an individual effect, else that problem would get even more complicated.

There's even more ambiguity when we consider that we can change the OsbLayer in the Layers Menu. Now that layer I declared in the effect's code may not even be in ``OsbLayer.Background``, but now in ``OsbLayer.Passing``, because I set it that way in the UI.

I'm trying to think of alternative solutions to this, especially considering the example code I wrote in the OP, but they seem to unravel more complications. If the passed argument overwrites an existing layer's place (i.e. ``slayer``'s ``OsbLayer.Background`` takes precedence because it's later), would that cause a problem with user-defined Layer menu choices, because both would be setters? I also don't think treating the two ``GetLayer`` instances as different layers is a good idea either since that's confusing.

Do you have any thoughts of how you'd rather see it being treated? Overall, I find these drawbacks parallel to how default values are treated with configurables -- that the passed ``OsbLayer`` is like an initial or default value if new. I'm not sure if I can see any simpler or intuitive way of asking to set an initial OsbLayer from a script aside from what I have right now.

Thanks for checking back on this. Apologies for writing walls of text overexplaining everything. I don't know if your beloved parasite is fond of walls of text. 💩